src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -3464,7 +3464,7 @@ class CreateStatementSegment(BaseSegment):
         ),
         # Next set are Storage Integration statements
         # https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration.html
-        AnySetOf(
+        AnyNumberOf(
```

Comment:
Check out the existing `snowflake/snowflake_create_storage_integration.sql` test case. It has the following statement in it:

```
create storage integration azure_int
  type = external_stage
  storage_provider = azure
  enabled = true
  azure_tenant_id = '<tenant_id>'
  storage_allowed_locations = ('azure://myaccount.blob.core.windows.net/mycontainer/path1/', 'azure://myaccount.blob.core.windows.net/mycontainer/path2/');
```

It has `azure_tenant_id = ...` and `storage_provider = AZURE`. Both of which are matched by `AzureStorageIntegrationParameters`. If it's set to `AnySetOf`, parsing fails because it can only use `AzureStorageIntegrationParameters` once, and that first use is on `storage_provider = azure` - therefore it can't use it again when it comes to `azure_tenant_id = '<tenant_id>'`.

I agree it feels gross, but it feels like the tidiest solution. Any better ideas?

Reply:
Ah okay, it looks like this was just not put together right in the first place. What we may want to do is move those 3 into private properties of the class as a list of options, then add them with a list unpack to the grammar. `_azure_params = [Sequence(), Sequence()] ... match_grammar = AnySetOf(other_things, *_azure_params)

Reply:
Hmmmm - good idea 👍